task.go

```diff
@@ -294,25 +293,27 @@ func safeParseSource(f string) (pkg *ast.Package, err error) {
 		}
 	}()
 
-	pkg = parser.ParseSource(f)
-	return pkg, err
+	return parser.Parse(f)
 }
 
 // UpdateFlux updates the TaskUpdate to go from updating options to updating a flux string, that now has those updated options in it
 // It zeros the options in the TaskUpdate.
-func (t *TaskUpdate) UpdateFlux(oldFlux string) (err error) {
+func (t *TaskUpdate) UpdateFlux(parser FluxLanguageService, oldFlux string) (err error) {
 	if t.Flux != nil && *t.Flux != "" {
 		oldFlux = *t.Flux
 	}
 	toDelete := map[string]struct{}{}
-	parsedPKG, err := safeParseSource(oldFlux)
+	if parser == nil {
+		return &Error{
+			Code: EInternal,
+			Msg:  "flux parser is not configured; updating a task requires the flux parser to be set",
+		}
+	}
+	parsedPKG, err := safeParseSource(parser, oldFlux)
```

Comment:
Is this the only place where we have to check that the parser is properly configured?

Reply:
No, I think we probably should check it in other locations. It just panics in those places if it isn't configured. I will add that before merging.